homeassistant/components/alarm_control_panel/group.py

```diff
@@ -0,0 +1,241 @@
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def code_format(self):
+        """The code format is customizable."""
+        return self._code_format
+
+    @asyncio.coroutine
+    def async_alarm_disarm(self, code=None):
+        """Send disarm command."""
+        yield from self._async_send_message(SERVICE_ALARM_DISARM,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_home(self, code=None):
+        """Send arm home command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_HOME,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_away(self, code=None):
+        """Send arm away command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_AWAY,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_night(self, code=None):
+        """Send arm night command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_NIGHT,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_trigger(self, code=None):
+        """Send alarm trigger command."""
+        yield from self._async_send_message(SERVICE_ALARM_TRIGGER,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_custom_bypass(self, code=None):
+        """Send arm custom bypass command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_CUSTOM_BYPASS,
+                                            code=code)
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        # _update_state always behaves as if a transition was in progress.
+        # But if the pre- and post-state match, we're not "pending"!
+        self._update_state()
+        if self._pre_state == self._post_state:
+            self._state = self._post_state
+        else:
+            self._state = STATE_ALARM_PENDING
+        return self._state
+
+    def _update_state(self):
+        """Update the current transition state of the device."""
+        def _common_state(states):
+            # This can happen if the sub-panels have not loaded yet.
+            if len(states) == 0:
+                return STATE_ALARM_DISARMED
+
+            # One triggered sub-panel is enough to trigger the whole panel.
+            if states[STATE_ALARM_TRIGGERED]:
+                return STATE_ALARM_TRIGGERED
+
+            # This only happens if a member does not provide pre or
+            # post_pending_state.  In that case, ensure the pending state
+            # percolates up to device_state_attributes, and the attribute
+            # will be left out in the group as well.
+            if states[STATE_ALARM_PENDING]:
+                return STATE_ALARM_PENDING
+
+            # If all entities have the same state, great.  Otherwise,
```

Comment:
There's two spaces after the period, before "Otherwise". Please remove one space.